protocol/src/lib.rs

```diff
@@ -570,12 +573,19 @@ pub struct WriteHeader {
     pub contexts: Vec<BlockContext>,
 }
 
+#[derive(Debug, PartialEq, Copy, Clone, Serialize, Deserialize)]
+pub enum ReadBlockContext {
```

Comment:
I really like this stronger type over the Option

Reply:
Me too 😀